sopel/privileges.py

```diff
@@ -1,146 +1,147 @@
-"""Constants for user privileges in channels.
+"""Constants for user privileges in channels."""
+# Copyright 2023 dgw, technobabbl.es
+from __future__ import annotations
 
-Privilege levels
-================
+import enum
 
-Historically, there were two user privileges in channels:
 
-* :data:`OP`: channel operator, or chanop, set and unset by ``+o`` and ``-o``
-* :data:`VOICE`: the privilege to send messages to a channel with the
-  ``+m`` mode, set and unset by ``+v`` and ``-v``
+__all__ = [
+    'AccessLevel',
+]
 
-Since then, other privileges have been adopted by IRC servers and clients:
 
-* :data:`HALFOP`: intermediate level between Voiced and OP, set and unset by
-  ``+h`` and ``-h``
-* :data:`ADMIN`: channel admin, above OP and below OWNER, set and unset by
-  ``+a`` and ``-a``
-* :data:`OWNER`: channel owner, above ADMIN and OP, set and unset by ``+q`` and
-  ``-q``
+class AccessLevel(enum.IntFlag):
+    """Enumeration of available user privilege levels.
 
-.. important::
+    Comparing privileges
+    ====================
 
-    Not all IRC networks support these added privilege modes. If you are
-    writing a plugin for public distribution, ensure your code behaves sensibly
-    if only +v (voice) and +o (op) modes exist.
+    This class represents privileges as powers of two, with higher values
+    assigned to higher-level privileges::
 
-Compare privileges
-==================
+        >>> from sopel.privileges import AccessLevel
+        >>> AccessLevel.VOICE < AccessLevel.HALFOP < AccessLevel.OP \\
+        ... < AccessLevel.ADMIN < AccessLevel.OWNER
+        True
 
-This module represents privileges as powers of two, with higher values assigned
-to higher-level privileges::
+    Then a user's privileges are represented as a sum of privilege levels::
 
-    >>> from sopel.privileges import VOICE, HALFOP, OP, ADMIN, OWNER
-    >>> VOICE < HALFOP < OP < ADMIN < OWNER
-    True
+        >>> AccessLevel.VOICE
+        1
+        >>> AccessLevel.OP
+        4
+        >>> priv = AccessLevel.VOICE | AccessLevel.OP
+        >>> priv
+        5
 
-Then a user's privileges are represented as a sum of privilege levels::
+    This allows using comparators and bitwise operators to compare privileges.
+    Here, ``priv`` contains both VOICE and OP privileges, but not HALFOP::
 
-    >>> VOICE
-    1
-    >>> OP
-    4
-    >>> priv = VOICE | OP
-    >>> priv
-    5
+        >>> priv >= AccessLevel.OP
+        True
+        >>> bool(priv & AccessLevel.HALFOP)
+        False
 
-This allows to use comparators and bitwise operators to compare privileges::
+    .. important::
 
-    >>> priv >= OP
-    True
-    >>> bool(priv & HALFOP)
-    False
+        Do not refer directly to the integer value of a privilege level in your
+        code; the values may change. Use the appropriate member of this class as
+        a reference point instead.
```

Comment:
I think this is part of my ambivalence about describing the _powers of two_ implementation: it reads to me as nudging users a bit more towards thinking about flags as integers.

Reply:
It's a hedge against Sphinx showing the actual value of each enum member in generated docs. I'm not sure how to turn it off, but would it help to omit the "integer" reference here and only talk about not hard-coding flag values (of an unspecified type)?

Reply:
That would resolve my ambivalence about the wording, yea